falcon/api.py

```diff
@@ -532,18 +533,27 @@ def add_error_handler(self, exception, handler=None):
         the client.  Alternatively, a handler may modify `resp`
         directly.
 
-        Error handlers are matched in LIFO order. In other words, when
-        searching for an error handler to match a raised exception, and
-        more than one handler matches the exception type, the framework
-        will choose the one that was most recently registered.
-        Therefore, more general error handlers (e.g., for the
-        standard ``Exception`` type) should be added first, to avoid
-        masking more specific handlers for subclassed types. For example::
+        An error handler "matches" a raised exception if the exception is an
+        instance of the corresponding exception type. If more than one error
+        handler matches the raised exception, the framework will choose the
+        most specific one. If multiple error handlers are registered for the
+        *same* exception class, then the most recently-registered handler is
+        used. For example, suppose we register error handlers as follows::
 
             app = falcon.API()
-            app.add_error_handler(Exception, custom_handle_uncaught_exception)
+            app.add_error_handler(falcon.HTTPNotFound, custom_handle_not_found)
             app.add_error_handler(falcon.HTTPError, custom_handle_http_error)
-            app.add_error_handler(CustomException)
+            app.add_error_handler(Exception, custom_handle_uncaught_exception)
+            app.add_error_handler(falcon.HTTPNotFound, custom_handle_404)
+
+        If an instance of ``falcon.HTTPForbidden`` is raised, it will be
```

Comment:
Open questions on the style in this section:

* Should I use single or double backticks for the handler function names? I couldn't decide whether they counted as "variables" or "inline code".
 * Should the ussages of falcon.HTTP* in this example include :class: so that there's a link to the relevant part of the api doc? I didn't do that to start because the specific error classes used for the examples are fairly arbitrary.

Reply:
double backticks are fine, although I tend to add parens to the function name myself to help make it obvious what it is. 

Re class names, I'm fine either way TBH, since you are referencing example code here. But it couldn't hurt to add the `:class:` directive.